qiskit_addon_sqd/fermion.py

```diff
@@ -302,6 +302,9 @@ def diagonalize_fermionic_hamiltonian(
                 raw_bitstrings, raw_probs, current_occupancies, n_alpha, n_beta, rand_seed=rng
             )
 
+        if bitstrings.size == 0:
+            raise ValueError("No valid bitstrings for subsample.")
```

Comment:
Actually, let's move this check to inside the `if current_occupancies is None:` block, after line 297. I believe this is the only place it can be triggered. Then I think we can improve the error message, to something like
>The input bit array did not contain any valid bitstrings. Either pass a bit array that contains at least one valid bitstring (with the correct right and left Hamming weights), or specify a value for initial_occupancies.